library/ssl_tls13_client.c

```diff
@@ -1153,6 +1147,29 @@ int mbedtls_ssl_tls13_write_client_hello_exts( mbedtls_ssl_context *ssl,
     }
 #endif
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+    if( mbedtls_ssl_conf_tls13_some_psk_enabled( ssl ) &&
+        ( mbedtls_ssl_conf_has_static_psk( ssl->conf ) == 1 ||
```

Comment:
```suggestion
```
I think we discussed that we will support only early data through tickets, at least to start with. Otherwise as mentioned in #6326 I think there are other checks to do: the TLS version associated to the ticket, ...

Reply:
I am not sure why should we only support tickets firsly, seems with openssl server, the early data indication test can go smoothly.  Anyway, I will add one guard to the psk supported, which default to be **disabled**

Reply:
The description from RFC8446 section 4.10: 
Client side: **When a PSK is used** and **early data is allowed for that PSK**, the client
   can send Application Data in its first flight of messages.  If the
   client opts to do so, it MUST supply both the "pre_shared_key" and
   "early_data" extensions.

Server side:
In order to accept early data, the server MUST have accepted **a PSK
   cipher suite** and **selected the first key offered** in the client's
   "pre_shared_key" extension.  In addition, it MUST **verify** that the
   following values are the same as those associated with the
   selected PSK:

   -  **The TLS version number**

   -  **The selected cipher suite**

   -  **The selected ALPN** [[RFC7301](https://datatracker.ietf.org/doc/html/rfc7301)] protocol, if any
   
Seems in client side, it doesn't refer the version number and cipher cuite verification, should we add them?

Reply:
For time being, I prefer keep it. That can make the development easier, although PSK has replay risk. 
Early data with session ticket, many feature is missing in `prototype` also. For example, serialization session ticket does not include `max_early_data_size`. 

I suggestion add PSK firstly with a configuration option that should be disabled by default.